src/main/java/core/basesyntax/service/impl/ServiceStrategy.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class ServiceStrategy {
+    public OperationHandler getServiceStrategy(FruitTransaction transaction) {
+        switch (transaction.getOperation()) {
+            case BALANCE :
+                return new BalanceHandler();
+            case SUPPLY :
+                return new SupplyHandler();
+            case PURCHASE :
+                return new PurchaseHandler();
+            default :
+                return new ReturnHandler();
```

Comment:
it's better to create a map where key will be type of operation and value handler and pass this map to the constructor of this class